tests/benchmark/stateful/bloatnet/depth_benchmarks/AttackOrchestrator.sol

```diff
@@ -0,0 +1,94 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title AttackOrchestrator
+ * @dev Orchestrates attacks on CREATE2-deployed contracts by deriving their addresses
+ *      and calling attack(uint256) on each one within a specified range.
+ */
+contract AttackOrchestrator {
+    // Immutable storage for gas efficiency
+    address public immutable deployerAddress;
+    bytes32 public immutable initCodeHash;
+
+    // Attack function selector for attack(uint256)
+    bytes4 private constant ATTACK_SELECTOR = 0x64dd891a;
+
+    /**
+     * @dev Constructor sets the deployer address and init code hash
+     * @param _deployer The address that deployed the target contracts via CREATE2
+     * @param _initCodeHash The keccak256 hash of the target contract's init code
+     */
+    constructor(address _deployer, bytes32 _initCodeHash) {
+        deployerAddress = _deployer;
+        initCodeHash = _initCodeHash;
+    }
+
+    /**
+     * @dev Executes attacks on a range of CREATE2-deployed contracts
+     * @param value The value to pass to each attack() call
+     * @param startIndex The starting index (inclusive)
+     * @param endIndex The ending index (exclusive)
+     */
+    function attack(uint256 value, uint256 startIndex, uint256 endIndex) external {
+        // Load immutables into local variables for assembly access
+        address _deployer = deployerAddress;
+        bytes32 _codeHash = initCodeHash;
+
+        // Use assembly for gas-efficient CREATE2 address derivation and calls
+        assembly {
+            // Use local variables instead of immutables
+            let deployer := _deployer
+            let codeHash := _codeHash
+
+            // Loop through the range
+            for { let i := startIndex } lt(i, endIndex) { i := add(i, 1) } {
+                // Derive CREATE2 address
+                // Formula: keccak256(0xff ++ deployer ++ salt ++ initCodeHash)[12:]
+
+                // Store CREATE2 preimage in memory
+                // The layout should be: 0xff (1 byte) + deployer (20 bytes) + salt (32 bytes) + codeHash (32 bytes)
+                // Total: 85 bytes
+                let memPtr := 0x00
+                mstore8(memPtr, 0xff)                                    // 0xff prefix at byte 0
```

Comment:
There are constants stored in this for loop which do not need to be there. You can move these constant mstores before the loop (to init the constant parts of memory once)